linkerd/app/admin/src/server.rs

```diff
@@ -82,6 +83,52 @@ impl<M> Admin<M> {
             .expect("builder with known status code must not fail")
     }
 
+    fn env_rsp<B>(req: Request<B>) -> Response<Body> {
+        use std::{collections::HashMap, env, ffi::OsString};
+
+        if req.method() != http::Method::GET {
+            return Self::method_not_allowed();
+        }
+
+        if let Err(not_acceptable) = json::accepts_json(&req) {
+            return not_acceptable;
+        }
+
+        fn unicode(s: OsString) -> String {
+            s.to_string_lossy().into_owned()
+        }
+
+        let query = req
+            .uri()
+            .path_and_query()
+            .and_then(http::uri::PathAndQuery::query);
+        let env = if let Some(query) = query {
+            if query.contains('=') {
+                return json::json_error_rsp(
+                    "env.json query parameters may not contain key-value pairs",
+                    StatusCode::BAD_REQUEST,
+                );
+            }
```

Comment:
@olix0r is this how we want to do the query params, or would we rather do something like `?env_var=FOO&env_var=BAR`?

Reply:
This seems good to me.